R_scripts/licordat.R

```diff
@@ -16,12 +16,14 @@ read_licor_data <- function(filename) {
   date <- file[which(grepl("^Type", file)) + 1]
   temp20 <- file[grepl("^Comments:", file)]
 
+  # Find beginning and end indices of raw data for each measurment
   tablestarts <- grep("^Type", file)
   tablestops <- grep("^CrvFitStatus", file)
+  
+  # Average values and place in matrix to be added to final df
   tcham <- matrix()
   t5 <- matrix()
-  
-  for (i in 1:length(tablestarts)) {
+  for (i in seq_along(tablestarts)) {
     df <- read.table(filename, skip = tablestarts[i] - 1, header = TRUE, 
                nrows = tablestops[i] - tablestarts[i] - 1)
     tcham[i] <- round(mean(df$Pressure), digits = 2)
```

Comment:
Here, the data frame **df** splits the Date column date and time into separate columns, causing the data to shift to the right one column. I used the df$Pressure notation as a placeholder since it works for now. Any ideas on how to separate the values? I know _read.table()_ has a "sep" argument.

Reply:
Ah, interesting, this was subtle. In the file, the various fields are separated by tab characters; but the `read.table` call above uses the default, which is separate on _any_ whitespace. So it splits apart the date/time entry into separate columns, throwing everything off by 1.

The solution (I think, haven't tested) is to tell `read.table` to split on tabs only:
```R
     df <- read.table(filename, skip = tablestarts[i] - 1, header = TRUE, sep = "\t",
```

Reply:
Ah, that should work. I'm getting an error, though, because there is an extra column heading "Annotation", so it still matches up even though it's incorrect. I can explain this better if I show you the data setup tomorrow, if possible. But it seems like I'd need to delete that extra column heading or fill in with N/As.

Reply:
```R
     df <- read.table(filename, skip = tablestarts[i] - 1, header = TRUE, sep = "\t", fill = TRUE,
 ```